backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/search/predicate/impl/SearchPredicateFactoryImpl.java

```diff
@@ -72,14 +80,35 @@ else if ( !selectedFormatter.equals( formatter ) ) {
 			}
 		}
 		if ( selectedFormatter == null ) {
-			throw log.unknownFieldForSearch(
-					absoluteFieldPath,
-					getIndexNames()
-					);
+			throw log.unknownFieldForSearch( absoluteFieldPath, getIndexNames() );
 		}
 		return selectedFormatter;
 	}
 
+	private void checkNestedField(String absoluteFieldPath) {
+		boolean found = false;
+
+		for ( ElasticsearchIndexModel indexModel : indexModels ) {
+			ElasticsearchObjectNodeModel nodeModel =
+					indexModel.getObjectNodeModel( absoluteFieldPath );
```

Comment:
From an API POV, I would have thought of:
```
indexModel.getNodeModel( absoluteFieldPath ).isObject();
```
(or maybe even `isNested()`)

Wondering if it's really worth it to have 2 separate maps for them.

Reply:
The thing is, `ElasticsearchObjectNodeModel` and `ElasticsearchFieldModel` have very few in common. This `isObject` method, and maybe a `getParent` method, but that's all.
If I did what you're suggesting, `getNodeModel` would have to return some interface with basically no useful method, and I would have to cast the interface somehow. Not sure it would be much cleaner...
I'll think it over, maybe I can find some nicer way to do it than this multiple-for garbage...
Also, this is not API. Not even SPI.

Reply:
Yeah, I was not using API as in API/SPI.

I hear you. It just feels weird.